cmd/helm/outdated.go

```diff
@@ -0,0 +1,231 @@
+			if err != nil {
+				return err
+			}
+			return outfmt.Write(out, newOutdatedListWriter(releases, cfg, out, devel))
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.Bool("devel", false, "use development versions (alpha, beta, and release candidate releases), too. Equivalent to version '>0.0.0-0'.")
+	flags.BoolVarP(&client.Short, "short", "q", false, "output short (quiet) listing format")
+	flags.BoolVarP(&client.ByDate, "date", "d", false, "sort by release date")
+	flags.BoolVarP(&client.SortReverse, "reverse", "r", false, "reverse the sort order")
+	flags.BoolVarP(&client.All, "all", "a", false, "show all releases, not just the ones marked deployed or failed")
+	flags.BoolVar(&client.Uninstalled, "uninstalled", false, "show uninstalled releases")
+	flags.BoolVar(&client.Superseded, "superseded", false, "show superseded releases")
+	flags.BoolVar(&client.Uninstalling, "uninstalling", false, "show releases that are currently being uninstalled")
+	flags.BoolVar(&client.Deployed, "deployed", false, "show deployed releases. If no other is specified, this will be automatically enabled")
+	flags.BoolVar(&client.Failed, "failed", false, "show failed releases")
+	flags.BoolVar(&client.Pending, "pending", false, "show pending releases")
+	flags.BoolVarP(&client.AllNamespaces, "all-namespaces", "A", false, "list releases across all namespaces")
+	flags.IntVarP(&client.Limit, "max", "m", 256, "maximum number of releases to fetch")
+	flags.IntVar(&client.Offset, "offset", 0, "next release name in the list, used to offset from start value")
+	bindOutputFlag(cmd, &outfmt)
+
+	return cmd
+}
+
+type outdatedElement struct {
+	Name         string `json:"name"`
+	Namespace    string `json:"namespace"`
+	InstalledVer string `json:"installed_version"`
+	LatestVer    string `json:"latest_version"`
+	Chart        string `json:"chart"`
+}
+
+type outdatedListWriter struct {
+	releases []outdatedElement // Outdated releases
+}
+
+func newOutdatedListWriter(releases []*release.Release, cfg *action.Configuration, out io.Writer, devel bool) *outdatedListWriter {
+	outdated := make([]outdatedElement, 0, len(releases))
+
+	// we initialize the Struct with default Options but the 'devel' option
+	// can be set by the User, all the other ones are not relevant.
+	searchRepo := searchRepoOptions{
+		versions:     false,
+		regexp:       false,
+		devel:        devel,
+		maxColWidth:  50,
+		version:      "",
+		repoFile:     settings.RepositoryConfig,
+		repoCacheDir: settings.RepositoryCache,
+	}
+
+	// initialize Repo index first
+	index, err := initSearch(out, &searchRepo)
+	if err != nil {
+		// TODO: Find a better way to exit
+		fmt.Fprintf(out, "%s", errors.Wrap(err, "ERROR: Could not initialize search index").Error())
+		os.Exit(1)
+	}
+
+	results := index.All()
+	for _, r := range releases {
+		// search if it exists a newer Chart in the Chart-Repository
+		repoResult, err := searchChart(results, r.Chart.Name(), r.Chart.Metadata.Version, devel)
+		if err != nil {
```

Comment:
This cause the whole plugin to error if there's just one chart that was deployed from local `Could not find any Repo which contains ...`. Even if that's intentional, there should be a parameter to turn such a case into a warning.

Reply:
Done with commit 50f1deb1ed00e64dfa9aea964927a065a4a86fab